test/ratelimit.js

```diff
@@ -63,49 +63,42 @@ describe('ratelimit middleware', function() {
   });
 
   describe('id', function (done) {
-    var ids = ['id1', 'id2'];
-    var guard = 0;
-    var app;
-
-    var routeHitTimes = function (n) {
-      return function () {
-        guard.should.be.equal(n);
-      }
-    };
-
-    beforeEach(function(done) {
-      app = koa();
+    it('should allow specifying a custom `id` function', function (done) {
+      var app = koa();
 
       app.use(ratelimit({
-        duration: rateLimitDuration,
         db: db,
-        max: 2,
         id: function (ctx) {
-          ctx.should.be.ok;
-          return ids[guard++]
-        }
+          return ctx.request.header.foo;
+        },
+        max: 2
       }));
 
-      app.use(function* (next) {
-        this.body = goodBody + guard;
-      });
+      request(app.listen())
+        .get('/')
+        .set('foo', 'bar')
+        .expect(function(res) {
+          res.header['x-ratelimit-remaining'].should.equal('1');
+        })
+        .end(done);
+    });
 
-      guard = 0;
+    it('should not limit if `id` returns `false`', function (done) {
+      var app = koa();
 
-      setTimeout(function() {
-        request(app.listen())
-          .get('/')
-          .expect(200, goodBody + "1")
-          .expect(routeHitTimes(1))
-          .end(done);
-      }, rateLimitDuration);
-    });
+      app.use(ratelimit({
+        db: db,
+        id: function (ctx) {
+          return false;
+        },
+        max: 5
+      }));
 
-    it('should not limit when different ids', function (done) {
```

Comment:
`should not limit if id returns false` didn't cover ids are different, it just covered `id === false`.  i think we should better to add a test case for this situation.